spring-cloud-gcp-logging/src/main/java/com/google/cloud/spring/logging/aot/LoggingRuntimeHints.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.logging.aot;
+
+import com.google.cloud.spring.logging.LoggingAppender;
+import com.google.cloud.spring.logging.StackdriverJsonLayout;
+import com.google.cloud.spring.logging.TraceIdLoggingEnhancer;
+import java.util.Arrays;
+import org.springframework.aot.hint.MemberCategory;
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.RuntimeHintsRegistrar;
+import org.springframework.aot.hint.TypeReference;
+
+public class LoggingRuntimeHints implements RuntimeHintsRegistrar {
+  @Override
+  public void registerHints(RuntimeHints hints, ClassLoader classLoader) {
+    hints
+        .reflection()
+        .registerTypes(
+            Arrays.asList(
+                TypeReference.of(LoggingAppender.class),
+                TypeReference.of(TraceIdLoggingEnhancer.class),
+                TypeReference.of(StackdriverJsonLayout.class)),
+            hint ->
+                hint.withMembers(
+                    MemberCategory.INVOKE_DECLARED_CONSTRUCTORS,
+                    MemberCategory.INVOKE_PUBLIC_METHODS));
+  }
+}
```

Comment:
Out of curiosity, do we notice any advantages with using the RuntimeHintsRegistrar class over using static reflection configurations? 
Mostly in terms of choosing the static jsons vs a dynamic way of registering classes for reflection, I'm wondering if we will run into any breaking changes when we upgrade our Spring stack that introduces some changes to it's class definition.

Reply:
Good question! I went this way because it's [documented](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#native-image.advanced.custom-hints) and to be consistent to spring conventions, I notice spring-boot and spring-data projects are using this `RuntimeHintsRegistar` (e.g. [logging autoconfig](https://github.com/spring-projects/spring-boot/blob/6176eaa0189ee71638693079eb35fcd9dbb0d01f/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/logback/LogbackRuntimeHints.java))
Some other benefit include: this allows hints be tested; It's easy use spring concepts when registering hints (for example in [vision](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/5375e0c554b1c3c737e41889615c53478d58f6b4/spring-cloud-gcp-vision/src/test/java/com/google/cloud/spring/vision/nativeimage/TestRuntimeHints.java#L28-L29), this is for the test resources, but you can easily refer to spring resources when registering.)

Reply:
Thanks for the context! Considering this resolved to unblock this PR. Discussion has been moved offline.